WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java

```diff
@@ -474,41 +483,48 @@ private void approveComment() {
-                @Override
-                public void onActionResult(CommentActionResult result) {
-                    if (result != null && result.isSuccess()) {
-                        requestCompleted(ARG_ACTION_APPROVE);
-                    } else {
-                        requestFailed(ARG_ACTION_APPROVE);
-                    }
-                }
-            });
+            // Update pseudo comment (built from the note)
+            CommentModel comment = mNote.buildComment();
+            comment.setStatus(CommentStatus.APPROVED.toString());
+            SiteModel site = mSiteStore.getSiteBySiteId(mNote.getSiteId());
+            if (site == null) {
+                AppLog.e(T.NOTIFS, "Impossible to approve a comment on a site that is not in the App. SiteId: "
+                                   + mNote.getSiteId());
+                requestFailed(ARG_ACTION_APPROVE);
+                return;
+            }
 
+            // Push the comment
+            mDispatcher.dispatch(CommentActionBuilder.newPushCommentAction(new RemoteCommentPayload(site, comment)));
         }
 
-        private void replyToComment(){
-            if (mNote == null) return;
+        private void replyToComment() {
+            if (mNote == null) {
+                return;
+            }
 
             // Bump analytics
             AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_QUICK_ACTIONS_REPLIED_TO);
 
-
             if (!TextUtils.isEmpty(mReplyText)) {
-                CommentActions.submitReplyToCommentNote(mNote, mReplyText, new CommentActions.CommentActionListener() {
-                    @Override
-                    public void onActionResult(CommentActionResult result) {
-                        if (result != null && result.isSuccess()) {
-                            requestCompleted(ARG_ACTION_REPLY);
-                        } else {
-                            if (result != null && !TextUtils.isEmpty(result.getMessage())) {
-                                requestFailedWithMessage(result.getMessage(), true);
-                            } else {
-                                requestFailed(ARG_ACTION_REPLY);
-                            }
-                        }
-                    }
-                });
+                SiteModel site = mSiteStore.getSiteBySiteId(mNote.getSiteId());
+                if (site == null) {
+                    AppLog.e(T.NOTIFS, "Impossible to reply a comment on a site that is not in the App. SiteId: "
+                                       + mNote.getSiteId());
+                    requestFailed(ARG_ACTION_APPROVE);
+                    return;
+                }
+
+                // Pseudo comment (built from the note)
+                CommentModel comment = mNote.buildComment();
+
+                // Pseudo comment reply
+                CommentModel reply = new CommentModel();
```

Comment:
We had some discussion about this before, but I noticed you're not using `INSTANTIATE_COMMENT` at all in this PR.

I agree that it doesn't make much sense in `Note.buildComment` since it's already a `Note` and we don't want to add it to the `CommentModelTable`, but shouldn't we be using it here at least (also in `CommentDetailFragment.java`)?